extension/sqlpgq/sqlpgq_functions/sqlpgq_scan.cpp

```diff
@@ -0,0 +1,300 @@
+	for(shared_ptr<PropertyGraphTable> entry : vec) {
+		if(entry -> table_name == table_name) {
+			return entry;
+		}
+	}
+	throw BinderException("Table name %s does not exist", table_name);
+	return nullptr;
+}
+
+static void ScanPGVColFunction(ClientContext &context, TableFunctionInput &data_p, DataChunk &output) {
+	bool &gstate = ((CSRScanState &)*data_p.global_state).finished;
+
+        if(gstate) {
+                output.SetCardinality(0);
+                return;
+        }
+
+        gstate = true;
+
+        auto sqlpgq_state_entry = context.registered_state.find("sqlpgq");
+        if (sqlpgq_state_entry == context.registered_state.end()) {
+                //! Wondering how you can get here if the extension wasn't loaded, but leaving this check in anyways
+                throw InternalException("The SQL/PGQ extension has not been loaded");
+        }
+        auto sqlpgq_state = reinterpret_cast<SQLPGQContext *>(sqlpgq_state_entry->second.get());
+
+        auto pg_name = ((PGScanVColData *)(data_p.bind_data))->pg_name;
+	auto table_name = ((PGScanVColData *)(data_p.bind_data))->table_name;
+        auto pg = sqlpgq_state->GetPropertyGraph(pg_name);
+
+	auto table_entry = find_table_entry(pg->vertex_tables, table_name);
+	
+	output.data[0].SetVectorType(VectorType::FLAT_VECTOR);
+        auto colsdata = FlatVector::GetData<string_t>(output.data[0]);
+        idx_t size = 0;
+        for(auto ele : table_entry -> column_names) {
+                colsdata[size] = string_t(ele.c_str(), ele.size());
+                size++;
+        }
+        output.SetCardinality(size);
+}
+
+static void ScanPGEColFunction(ClientContext &context, TableFunctionInput &data_p, DataChunk &output) {
+        bool &gstate = ((CSRScanState &)*data_p.global_state).finished;
+
+        if(gstate) {
+                output.SetCardinality(0);
+                return;
+        }
+
+        gstate = true;
+
+        auto sqlpgq_state_entry = context.registered_state.find("sqlpgq");
+        if (sqlpgq_state_entry == context.registered_state.end()) {
+                //! Wondering how you can get here if the extension wasn't loaded, but leaving this check in anyways
+                throw InternalException("The SQL/PGQ extension has not been loaded");
+        }
+        auto sqlpgq_state = reinterpret_cast<SQLPGQContext *>(sqlpgq_state_entry->second.get());
+
+        auto pg_name = ((PGScanEColData *)(data_p.bind_data))->pg_name;
+        auto table_name = ((PGScanEColData *)(data_p.bind_data))->table_name;
+        auto pg = sqlpgq_state->GetPropertyGraph(pg_name);
+
+        auto table_entry = find_table_entry(pg->edge_tables, table_name);
+
+        output.data[0].SetVectorType(VectorType::FLAT_VECTOR);
+        auto colsdata = FlatVector::GetData<string_t>(output.data[0]);
+        idx_t size = 0;
+        for(auto ele : table_entry -> column_names) {
```

Comment:
Think this can be `const auto &ele : table_entry->column_names` (also didn't know there can be spaces between the `->`)